File: RetailCalcPro/RetailCalcPro/Models/DiscountType.swift
import Foundation

enum DiscountType: Codable, Equatable, Hashable {
    case percentage(Int) // e.g., 10 means 10% off
    case amount(Int)     // e.g., 100 means 100 yen off

    var label: String {
        switch self {
        case .percentage(let value): return "\(value)%引"
        case .amount(let value): return "\(value)円引"
        }
    }

    var shortLabel: String {
        switch self {
        case .percentage(let value): return "\(value)%"
        case .amount(let value): return "\(value)円"
        }
    }
}

File: android/app/src/main/java/com/retailcalcpro/app/model/DiscountType.kt
package com.retailcalcpro.app.model

sealed class DiscountType {
    data class Percentage(val value: Int) : DiscountType()
    data class Amount(val value: Int) : DiscountType()

    val label: String
        get() = when (this) {
            is Percentage -> "${value}%引"
            is Amount -> "${value}円引"
        }

    val shortLabel: String
        get() = when (this) {
            is Percentage -> "${value}%"
            is Amount -> "${value}円"
        }
}
